src/solvers/sat/satcheck_minisat2.h

```diff
@@ -23,8 +23,14 @@ class Solver; // NOLINT(readability/identifiers)
 class SimpSolver; // NOLINT(readability/identifiers)
 }
 
+class optionst;
+
 struct satcheck_minisat_infot final
 {
+  /// Parse satcheck options
+  /// \param options Options object
+  /// \return An instance of satcheck_minisat_info object
+  static satcheck_minisat_infot parse(const optionst &options);
```

Comment:
How about making this a constructor (in addition to a default constructor)?

Reply:
I wanted to use a function, but assigning a function pointer behind an ifdef proved to be a little too cumbersome.

Reply:
I'm sure you've got good reasons for not doing it, but I don't really understand your explanation as I don't know which function pointer or ifdef that would be?

Reply:
The one in satcheck.h which switches between solvers. I'd have to create another one in .cpp file I think.

Reply:
How about declaring the interface in satcheck.h and then provide an implementation in each solver? Thus different solvers could happily parse/ignore different options? Maybe that's a bad idea for other reasons?

Reply:
I think it's "beyond the scope of this PR". The common interface isn't specified anywhere, macros just switch between different typedefs.